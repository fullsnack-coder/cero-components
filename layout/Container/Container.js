import React from 'react'
import PropTypes from 'prop-types'
import styles from './Container.module.css'
import withStyles from '../../hocs/withStyles'

export const Container = ({
  getStyles,
  children,
  isCentered,
  isPlayground,
}) => {
  return (
    <div
      className={getStyles('container', {
        'is-playground': isPlayground,
        'is-centered': isCentered,
      })}
    >
      {children}
    </div>
  )
}

Container.propTypes = {
  children: PropTypes.node.isRequired,
  getStyles: PropTypes.func.isRequired,
  isPlayground: PropTypes.bool,
  isCentered: PropTypes.bool,
}

Container.defaultProps = {
  isPlayground: false,
  isCentered: true,
  getStyles: () => {},
}

export default withStyles(styles)(Container)
