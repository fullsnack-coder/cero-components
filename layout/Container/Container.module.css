.container {
  display: flex;
  width: 100%;
  max-width: var(--container-max-width);
  height: 100%;
  flex: auto;
  padding: 20px 15px 20px;
  margin: 0 auto;
}

.is-centered {
  justify-content: center;
}

.is-playground {
  border: var(--border-width-thick) dashed var(--color-primary);
}
