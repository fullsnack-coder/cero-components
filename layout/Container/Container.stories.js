import { Container, styles } from '.'

import { getTemplate } from '@glrodasz/storybook-tools-helpers'

const Template = getTemplate(Container, styles)

export default {
  title: 'Layout/Container',
  component: Container,
  args: {
    isPlayground: true,
    children: `Farthings pulled strain? Half they'll greater land sordid Elessar`,
  },
  argTypes: {
    children: { control: 'text' },
  },
}

export const Default = Template.bind({})

export const Centered = Template.bind({})
Centered.args = {
  isCentered: true,
}
