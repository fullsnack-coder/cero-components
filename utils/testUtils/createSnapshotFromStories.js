import React from 'react'
import { render } from '@testing-library/react'
import { composeStories } from '@storybook/testing-react'

// TODO: Move this to a generic approach where it
// reads all the stories and works as Storyshoot
const createSnapshotFromStories = (componentName, stories) => {
  const composedStories = composeStories(stories)

  Object.entries(composedStories).forEach(([story, Component]) => {
    describe(`when '${componentName}' ${story} is mounted`, () => {
      it('should render', () => {
        const { asFragment } = render(<Component />)
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
}

export default createSnapshotFromStories
