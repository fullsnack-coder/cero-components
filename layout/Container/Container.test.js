import * as stories from './Container.stories'
import createSnapshotFromStories from '../../utils/testUtils/createSnapshotFromStories'

describe('[ layout / Container ]', () => {
  describe('#Container', () => createSnapshotFromStories('Container', stories))
})
