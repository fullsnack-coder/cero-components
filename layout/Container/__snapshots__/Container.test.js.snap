// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ layout / Container ] #Container when 'Container' Centered is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="container is-playground is-centered"
  >
    Farthings pulled strain? Half they'll greater land sordid Elessar
  </div>
</DocumentFragment>
`;

exports[`[ layout / Container ] #Container when 'Container' Default is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="container is-playground"
  >
    Farthings pulled strain? Half they'll greater land sordid Elessar
  </div>
</DocumentFragment>
`;
